packages/db/src/models/granule.ts

```diff
@@ -5,30 +5,79 @@ import { tableNames } from '../tables';
 
 import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 
+import { GranuleExecutionHistoryPgModel } from './granule-execution-history';
+
 export default class GranulePgModel extends BasePgModel<PostgresGranule, PostgresGranuleRecord> {
   constructor() {
     super({
       tableName: tableNames.granules,
     });
   }
 
-  upsert(
+  async createWithExecutionHistory(
+    knexOrTrx: Knex | Knex.Transaction,
```

Comment:
Should this ever happen outside of a transaction?      The rest of our model operations are all generally single query operations and so having the transaction *or* knex object makes sense, but we don't want a situation where the granule is created, but the link to the execution fails.   I would think we'd want this to *only* accept Knex.Transaction.

Reply:
Also I'm wondering if the question at L 74 implies that we should be instead relying on logic in the caller to handle the  granuleExecutionHistoryPgModel.create call.

Reply:
Hmm yeah you're right that maybe we should only support a transaction for these methods handling multiple writes.

As for whether this model should even be writing the history records, see https://github.com/nasa/cumulus/pull/2091/files#r577969012